### ISSUE TITLE
Improve code coverage

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#1790]
Our project code coverage start to decrease with new features and reworked tests. We are now ~95% but were 96% recently and around ~98% the last year...

The idea is to identify the uncovered parts of code inspecting here, for example: https://codecov.io/gh/pytorch/ignite/tree/master/ignite and draft a PR with new tests.

Some of new test can be very simple and others could be very sophisticated to enter a particular code line.
Let's improve the coverage of all modules, except `distributed` for instance.

**Comment by Moh-Yakoub** [comment]
That sounds like a group effort. shall we update the task with a checkbox list of packages so that we make sure that no two people will be working on the same package?

**Comment by soma2000-lang** [comment]
@vfdev-5  Can I work on ignite/engine/engine.py?

**Comment by soma2000-lang** [comment]
@vfdev I will try

**Comment by soma2000-lang** [comment]
@vfdev-5  https://app.codecov.io/gh/pytorch/ignite/blob/master/ignite/engine/engine.py in this file I see almost everything is covered .Please correct me if I am wrong

**Comment by vfdev-5** [comment]
@soma2000-lang yes 97% of the code is covered but we need to cover up to 100% if possible.

**Comment by soma2000-lang** [comment]
Ok

**Comment by mohameds-dev** [comment]
Hi, I am a senior CS student with industry experience in Python (Django) as well as my own side projects development.

I've practiced Test-Driven Development (TDD) as part of a Software Design class and I am familiar with how test coverage works in Python.

I am looking to practice open-source contribution by contributing to this project, specifically attempting to push the coverage to 100% [on this file as shown here](https://app.codecov.io/gh/pytorch/ignite/blob/master/ignite%2Fengine%2Fengine.py).

Could you assign me to this issue?
